lib/generators/declare_schema/migration/migrator.rb

```diff
@@ -451,31 +451,68 @@ def change_indexes(model, old_table_name, to_rename)
               end
           end
 
-          drop_indexes = (existing_indexes_without_primary_key - model_indexes_without_primary_key).map do |i|
+          indexes_to_drop = existing_indexes_without_primary_key - model_indexes_without_primary_key
+          indexes_to_add = model_indexes_without_primary_key - existing_indexes_without_primary_key
+
+          renamed_indexes_to_drop = []
+          renamed_indexes_to_add = []
+
+          indexes_to_drop.each { |index_to_drop|
+            renamed_columns = index_to_drop.columns.map do |column|
+              to_rename[column]
+            end
+
+            indexes_to_add.each { |index_to_add|
+              if renamed_columns == index_to_add.columns
+                renamed_indexes_to_drop.append(index_to_drop)
+                renamed_indexes_to_add.append(index_to_add)
+              end
+            }
+          }
+
+          drop_indexes = (indexes_to_drop - renamed_indexes_to_drop).map do |i|
             ::DeclareSchema::SchemaChange::IndexRemove.new(new_table_name, i.columns, unique: i.unique, where: i.where, name: i.name)
           end
 
-          add_indexes = (model_indexes_without_primary_key - existing_indexes_without_primary_key).map do |i|
+          add_indexes = (indexes_to_add - renamed_indexes_to_add).map do |i|
```

Comment:
Here is where I subtract out the indexes that I don't actually want to add. I do the same thing for the index removes.